tgui/packages/tgui/interfaces/CryopodConsole.js

```diff
@@ -0,0 +1,91 @@
+import { useBackend } from '../backend';
+import { Button, Section, NoticeBox, LabeledList, Collapsible } from '../components';
+import { Window } from '../layouts';
+
+export const CryopodConsole = (props, context) => {
+  const { data } = useBackend(context);
+  const { account_name, allow_items } = data;
+
+  return (
+    <Window title="Cryopod Console">
+      <Window.Content>
+        <Section title={`Hello, ${account_name || '[REDACTED]'}!`}>
+          This automated cryogenic freezing unit will safely store your
+          corporeal form until your next assignment.
+        </Section>
+        <CrewList />
+        {!!allow_items && <ItemList />}
+      </Window.Content>
+    </Window>
+  );
+};
+
+const CrewList = (props, context) => {
+  const { data } = useBackend(context);
+  const { frozen_crew } = data;
+
+  return (
+    <Collapsible title="Stored Crew">
+      {!frozen_crew.length ? (
+        <NoticeBox>No stored crew!</NoticeBox>
+      ) : (
+        <Section height={10} fill scrollable>
+          <LabeledList>
+            {frozen_crew.map((person) => (
+              <LabeledList.Item key={person} label={person.name}>
+                {person.rank}
+              </LabeledList.Item>
+            ))}
+          </LabeledList>
+        </Section>
+      )}
+    </Collapsible>
+  );
+};
+
+const ItemList = (props, context) => {
+  const { act, data } = useBackend(context);
+  const { frozen_items } = data;
+
+  const replaceItemName = (item) => {
+    let itemName = item.toString();
+    if (itemName.startsWith('the')) {
+      itemName = itemName.slice(4, itemName.length);
+    }
+    return itemName.replace(/^\w/, (c) => c.toUpperCase());
+  };
+
+  return (
+    <Collapsible title="Stored Items">
+      {!frozen_items.length ? (
+        <NoticeBox>No stored items!</NoticeBox>
+      ) : (
+        <>
+          <Section height={11} fill scrollable>
+            <LabeledList>
+              {frozen_items.map((item, index) => (
+                <LabeledList.Item
+                  key={item}
+                  label={replaceItemName(item)}
+                  buttons={
+                    <Button
+                      icon="arrow-down"
+                      content="Drop"
+                      mr={1}
+                      onClick={() => act('one_item', { item: index + 1 })}
```

Comment:
I'd be careful about passing the index as a parameter. It's possible for the index number in js and the location of the object in a DM list to not match up, especially when quickly dropping items or when the TGUI updates slowly. You should probably use some other unique ID system.

Reply:
I'm not sure how it would go wrong, as the DM list is only updated when pressing this button.

Reply:
My point is that, if the UI is displaying 7 elements, but by clicking a button you remove an element from the DM list resulting in the list only having 6 elements now. If the TGUI doesn't update in time, especially in cases of spam clicking, its possible for the TGUI to send an act looking for element 7 when it doesn't exist, or looking for element 5 when it's actually referring to element 4 but it's out of line because the index has not updated quick enough to match the new DM list passed to the TGUI.

Reply:
The out of bounds index references are handled in code, see https://github.com/ParadiseSS13/Paradise/pull/19540/files#diff-675e6667c9d8efb7b9611ff42d19b5e6c8a99a008146515e9f05f43f070dc0cfR101

The list order thing could be an issue, so I'll have a think about how to resolve this.